modules/drivers/mongo/src/metabase/driver/mongo/conversion.clj

```diff
@@ -37,7 +38,12 @@
 
   org.bson.types.Decimal128
   (from-document [^org.bson.types.Decimal128 input _opts]
-    (.bigDecimalValue input))
+    (try (.bigDecimalValue input)
+         ;; Convert negative and positive zero to 0M
+         (catch ArithmeticException e
+           (if (str/includes? (ex-message e) "zero can not be converted to a BigDecimal")
+             0M
+             (throw e)))))
```

Comment:
I'm really dismayed to see us pattern matching on an error message in production code.

- Can we call something on the `input` to check for this case?
- Is there any more structured information on the exception? Is it a custom subclass, or a straight `ArithmeticException`? Does it include a more specific `.cause` from the BSON library?

Reply:
I've removed exception handling in favor of `.equals`.

> Can we call something on the input to check for this case?

I'll keep in mind to examine different ways of value checks than exception handling first.

> Is there any more structured information on the exception?

It was `ArithmeticException` as follows:
```
#error {
 :cause "Negative zero can not be converted to a BigDecimal"
 :via
 [{:type java.lang.ArithmeticException
   :message "Negative zero can not be converted to a BigDecimal"
   :at [org.bson.types.Decimal128 bigDecimalValue "Decimal128.java" 297]}]
 :trace
...
```